### ISSUE TITLE
Math broken since 1.12.36 

### ISSUE THREAD

**Issue by rutchkiwi** [yogthos/Selmer#297]
Hi! 

I'm trying to upgrade selmer from my old .33 version. I tried to narrow the problem down by some bisecting, and when going to 1.12.36, "math" stopped working. An example:

```
(selmer/render "{{a-b}}" {:a 10 :b 5})
13:22:14.309 ERROR [] backoffice.handler - error when rendering template.
13:22:14.309 ERROR [] backoffice.handler - tag: {:tag-type :filter, :tag-value "a-b"}. Context was {:a 10, :b 5}
Execution error at backoffice.handler/missing-value-fn (handler.clj:55).
Selmer template error: {:tag-type :filter, :tag-value "a-b"} was null.
```

Can someone else reproduce this? I've got an error handler that throws exceptions when filters nil - this might work without that but I want to have that error handler.

```
(defn- missing-value-fn [tag context-map]
  (log/error "error when rendering template.")
  (log/errorf "tag: %s. Context was %s" tag context-map)
  (throw (Exception. (str "Selmer template error: " tag " was null."))))

(selmer.util/set-missing-value-formatter! missing-value-fn)
```

Note that this worked fine in 1.12.44, so I suspect https://github.com/yogthos/Selmer/pull/261 without having looked at the code in it.

**Comment by yogthos** [comment]
I think that might've been an incidental feature as there are no tests for it. I'm open to adding the functionality back in if you'd be ok to do a PR for the feature. Ideally, I'd like to retain the functionality @borkdude added with the whitespace PR as well.

**Comment by rutchkiwi** [comment]
Yeah I did notice that there was no docs for it, I was kinda hoping you wouldn't say that lol.
Thanks, then I know how it is. I think I might solve my issue using the `add` filter instead for now, there were only about ~5 occurances in my codebase of this thing. But would be nice to see if I can find time to add it back.
Super appreciate your lightning fast response @yogthos :)

**Comment by yogthos** [comment]
No prob, and I think you'd have to add a custom tag or a filter for this at the moment.

**Comment by yogthos** [comment]
you could actually add the functionality pretty easily using this library https://github.com/rm-hull/infix

```clojure
(require
  '[infix.core :refer [base-env]]
   '[jasentaa.parser :as p :refer [parse-all]]
   '[infix.grammar :refer [expression]])

(selmer/add-tag! :math
          (fn [args context-map]
            ((parse-all expression (apply str args))
             (merge base-env context-map))))

(selmer/render "{% math x + y * z %}" {:x 1 :y 2 :z 3})
```

I'd be open to a PR to add a new `:math` tag using infix if it works with Babashka.

**Comment by borkdude** [comment]
As discussed: I think it's better to keep this example as documentation (maybe add to the README) as how extensible Selmer is to do your own things, than bringing in extra dependencies for this (that also grow the size of babashka, increase load time of Selmer in the REPL, etc). This way people can use it without while Selmer can also stay light weight.

**Comment by borkdude** [comment]
Btw, the above example also works in babashka today :)

<img width="722" alt="Screenshot 2022-11-15 at 17 36 11" src="https://user-images.githubusercontent.com/284934/201975310-edf23988-7fd8-4195-8973-c8bca1dba538.png">

**Comment by yogthos** [comment]
I added an example in the readme [here](https://github.com/yogthos/Selmer#tags), @rutchkiwi if that works for your case we could probably close this? :)

**Comment by yogthos** [comment]
How this ended up working initially is definitely a fun puzzle to tease apart at some point.
